lxd/container_lxc.go

```diff
@@ -8717,6 +8781,77 @@ func (c *containerLXC) removeNetworkFilters() error {
 	return nil
 }
 
+// removeNetworkFilters removes any network level filters defined for the container.
+func (c *containerLXC) removeNetworkFilters(deviceName string, m types.Device) {
+	if m["hwaddr"] == "" {
+		logger.Error("Failed to remove network filters: ", log.Ctx{"container": c.Name(), "device": deviceName, "err": "hwaddr not defined"})
+		return
+	}
+
+	if m["host_name"] == "" {
+		logger.Error("Failed to remove network filters: ", log.Ctx{"container": c.Name(), "device": deviceName, "err": "host_name not defined"})
+		return
+	}
+
+	// Remove any IPv6 filters used for this container.
+	err := containerIptablesClear("ipv6", fmt.Sprintf("%s - ipv6_filtering", c.Name()), "filter")
+	if err != nil {
+		logger.Error("Failed to clear ip6tables ipv6_filter rules: ", log.Ctx{"container": c.Name(), "device": deviceName, "err": err})
+	}
+
+	// Get a current list of rules active on the host.
+	out, err := shared.RunCommand("ebtables", "-L", "--Lmac2", "--Lx")
+	if err != nil {
+		logger.Error("Failed to remove network filters: ", log.Ctx{"container": c.Name(), "device": deviceName, "err": err})
+		return
+	}
+
+	// Get a list of rules that we would have applied on container start.
+	rules := c.generateNetworkFilterEbtablesRules(m)
+
+	// Iterate through each active rule on the host and try and match it to one the LXD rules.
+	for _, line := range strings.Split(out, "\n") {
+		line = strings.TrimSpace(line)
+		fields := strings.Fields(line)
+		fieldsLen := len(fields)
+
+	RULES:
```

Comment:
We've so far avoided `goto` like syntax in the LXD codebase. It looks like in this case this could be done pretty easily by turning the rule matching logic into a function and calling it, avoiding the issue with the nested for loops.

Reply:
@stgraber I've added a function as suggested thanks